aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,387 @@
+    */
+  @transient private[this] lazy val defaultLabelInfo =
+    LabelsAndInfo(labelsInTrainingSet.indices, labelsInTrainingSet, Seq.empty, None)
+
+  /**
+    * Making from label to index into the sequence of all labels encountered during training.
+    */
+  private[this] val labelToInd: Map[K, Int] =
+    labelsInTrainingSet.zipWithIndex.map { case (label, i) => label -> i }(collection.breakOut)
+
+  override def subvalue(a: A): Subvalue[B, Map[K, Double]] = {
+    val li = labelsAndInfo(a, labelsOfInterest, labelToInd, defaultLabelInfo)
+
+    if (li.labels.isEmpty)
+      reportNoPrediction(modelId, li, auditor)
+    else {
+      val Features(x, missing, missingOk) = constructFeatures(a)
+
+      if (!missingOk)
+        reportTooManyMissing(modelId, li, missing, auditor)
+      else {
+        // TODO: To support label-dependent features, fill last parameter with a valid value.
+        val predictionTry = Try { predictor(x, li.labels, li.indices, sci.IndexedSeq.empty) }
+
+        predictionTry match {
+          case Success(pred) => reportSuccess(modelId, li, missing, pred, auditor)
+          case Failure(ex)   => reportPredictorError(modelId, li, missing, ex, auditor)
+        }
+      }
+    }
+  }
+
+  override def close(): Unit =
+    predictor match {
+      case closeable: Closeable => closeable.close()
+      case _ =>
+    }
+}
+
+object MultilabelModel extends ParserProviderCompanion {
+
+  /**
+    * Contains information about the labels to be used for predictions, and problems encountered
+    * while trying to get those labels.
+    * @param indices indices into the sequence of all labels seen during training.  These should
+    *                be sorted in ascending order.
+    * @param labels labels for which a prediction should be produced.  labels are parallel to
+    *               indices so `indices(i)` is the index associated with `labels(i)`.
+    * @param missingLabels a sequence of labels derived from the input data that could not be
+    *                      found in the sequence of all labels seen during training.
+    * @param problems any problems encountered when trying to get the labels.  This should only
+    *                 be present when the caller indicates labels should be embedded in the
+    *                 input data passed to the prediction function in the MultilabelModel.
+    * @tparam K type of label or class
+    */
+  protected[multilabel] case class LabelsAndInfo[K](
+      indices: sci.IndexedSeq[Int],
+      labels: sci.IndexedSeq[K],
+      missingLabels: Seq[K],
+      problems: Option[GenAggFuncAccessorProblems]
+  ) {
+    def missingVarNames: Seq[String] = problems.map(p => p.missing).getOrElse(Nil)
+    def errorMsgs: Seq[String] = {
+      missingLabels.map { lab => s"Label not in training labels: $lab" } ++
+      problems.map(p => p.errors).getOrElse(Nil)
+    }
+  }
+
+  private[multilabel] val NumLinesToKeepInStackTrace = 20
```

Comment:
This seems like a variable that is more general than just this class.  Is that so?

Reply:
Yeah.  But I don't really want to aggregate this right now.  I don't know exactly where it should go.  I need to think more about how to do this globally.